### PR TITLE
fix(styles): isolated point style overrides

### LINKED ISSUE #2277: Isolated point styles override user defined point styles
**Describe the issue**
With the changes from #2004 to expose isolated point styles, this inadvertently overrides provided styles such as `point.radius` only for the isolated points

**To Reproduce**
1. Go to https://codesandbox.io/s/romantic-tess-gwvxw4?file=/src/App.tsx
2. Notice the style difference between the isolated points (i.e. `null`, value, `null`).

**Expected behavior**
We should use merge the styles from `LineSeries.lineSeriesStyle.point` with `LineSeries.lineSeriesStyle.isolatedPoint`. Same for 
`AreaSeries` we should merge `AreaSeries.areaSeriesStyle.point` with `AreaSeries.areaSeriesStyle.isolatedPoint`.

**Screenshots**

romantic-tess-gwvxw4 - CodeSandbox 2023-12-08 at 12 53 33 PM

**Version (please complete the following information):**
 - Elastic Charts: [`v58.0.0`](https://github.com/elastic/elastic-charts/releases/tag/v58.0.0)

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/172995

**Checklist**
- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] Every related Kibana issue is listed under `Kibana Cross Issues` list and the `kibana cross issue` tag is applied

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

This change applies a hierarchy to the `(AreaSeriesStyles|LineSeriesStyles).isolatedPoint` styles. This is to prevent changing values in one to override the other. If the consumer applies a style to the `points` they would expect the style to transfer to all points.

The hierarchy would be (in order of highest priority):
- `(AreaSeriesStyles|LineSeriesStyles).isolatedPoint`
- `(AreaSeriesStyles|LineSeriesStyles).point`
  - This was the missing piece in the current logic
  - Skips `visible` override
- `Theme.(areaSeriesStyle|lineSeriesStyle).isolatedPoint`

## Details

I don't think the onus of this styling should be on the chart consumer to always supply a backup style to `isolatedPoint` whenever they are setting the `point` style.

## Issues

fix #2277

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated
- [x] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
- [x] Visual changes have been tested with `light` and `dark` themes

### CODE REVIEW

File: e2e/package.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
     "playwright": "playwright"
   },
   "dependencies": {
-    "@playwright/test": "^1.40.1",
-    "@types/node": "^20.10.0",
+    "@playwright/test": "^1.28.0",
```
[the remark this answers is not in the document]
Comment (nickofthyme) [reply e2e/package.json:15]: Oh thanks did't see that.

[the remark this answers is not in the document]
Comment (nickofthyme) [reply e2e/package.json]: I just updated playwright in #2296 cuz there were more complications and a large screenshot diff count.

**Comment by nickofthyme** [comment]
@markov00 could you take another look at this?

**Comment by nickofthyme** [comment]
Comment from https://github.com/elastic/elastic-charts/issues/2277#issuecomment-1860764785

> There is one more issue related to that:
> isolated points in Stacked areas are not shown if the consumer legitimately sets the areaPointStyle visibility to `false`. This with our default theme (EUI) isolated point on these time of charts are interactive but not visible:
> 
> Screenshot 2023-12-18 at 15 56 03
> 
> At this point, an `isolated point` should inherit the style of a point except for the visibility that should always be true, except if explicitly disabled in the isolated point style.
> 
>

**Review by markov00 — APPROVED** [review]
Seems to work fine for me, thanks!

**Pull request merged**